src/uwtools/drivers/mpas_init.py

```diff
@@ -0,0 +1,179 @@
+            symlinks[target] = linkname
+        yield [symlink(target=t, linkname=l) for t, l in symlinks.items()]
+
+    @tasks
+    def files_copied(self):
+        """
+        Files copied for run.
+        """
+        yield self._taskname("files copied")
+        yield [
+            filecopy(src=Path(src), dst=self._rundir / dst)
+            for dst, src in self._driver_config.get("files_to_copy", {}).items()
+        ]
+
+    @tasks
+    def files_linked(self):
+        """
+        Files linked for run.
+        """
+        yield self._taskname("files linked")
+        yield [
+            symlink(target=Path(target), linkname=self._rundir / linkname)
+            for linkname, target in self._driver_config.get("files_to_link", {}).items()
+        ]
+
+    @task
+    def namelist_file(self):
+        """
+        The namelist file.
+        """
+        fn = "namelist.init_atmosphere"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield None
+        stop_time = self._cycle + timedelta(
+            hours=self._driver_config["boundary_conditions"]["length"]
+        )
+        d = {
+            "nhyd_model": {
+                "config_start_time": self._cycle.strftime("%Y-%m-%d_%H:00:00"),
+                "config_stop_time": stop_time.strftime("%Y-%m-%d_%H:00:00"),
+            }
+        }
+        namelist = self._driver_config.get("namelist", {})
+        values = YAMLConfig(d)
+        values.update_values(namelist.get("update_values", {}))
+        namelist["update_values"] = values.data
+        self._create_user_updated_config(
+            config_class=NMLConfig,
+            config_values=namelist,
+            path=path,
+        )
+
+    @tasks
+    def provisioned_run_directory(self):
+        """
+        Run directory provisioned with all required content.
+        """
+        yield self._taskname("provisioned run directory")
+        yield [
+            self.boundary_files(),
+            self.files_copied(),
+            self.files_linked(),
+            self.namelist_file(),
+            self.runscript(),
+            self.streams_file(),
+        ]
+
+    @task
+    def runscript(self):
+        """
+        The runscript.
+        """
+        path = self._runscript_path
+        yield self._taskname(path.name)
+        yield asset(path, path.is_file)
+        yield None
+        self._write_runscript(path=path, envvars={})
+
+    @task
+    def streams_file(self):
+        """
+        The streams file.
+        """
+        fn = "streams.init_atmosphere"
+        yield self._taskname(fn)
+        path = self._rundir / fn
+        yield asset(path, path.is_file)
+        yield file(path=Path(self._driver_config["streams"]["path"]))
+        render(
+            input_file=Path(self._driver_config["streams"]["path"]),
+            output_file=path,
+            values_src=self._driver_config["streams"]["values"],
+        )
```

Comment:
Need to fill in `streams.init_atmosphere` file with appropriate values from the `yaml`.

Reply:
Are we assuming that the streams file is a Jinja2 template? IIRC that's not how it's supplied with MPAS. It's XML, right? If so, I guess I would have expected use of `lxml` (and probably its XPath support) to edit the file. I wouldn't hold up this PR for that, but I'd like to discuss this and potentially create a ticket for using `lxml` to edit the file that ships with MPAS. Apologies if I've misunderstood what's happening here or any assumptions we're making.

Reply:
I wouldn't be opposed to this approach for updating the XML that MPAS produces at build time. Right now, we've taken the one that's built with the mpas_app and turned it into a template. I think this would be relatively straight-forward to modify for support.

In general, it sounds like this could open us up to adding general XML to our config languages we support with YAML?

Reply:
General XML support could be added to the roadmap, but I don't think we'd need to go that far to directly support the MPAS XML files. I'd prefer to support MPAS in the short term, and consider general XML handling in the longer therm.

Reply:
Sounds good to me.

Reply:
I created [UW-556](https://jira.epic.oarcloud.noaa.gov/browse/UW-556) to address this.

Reply:
I also created [UW-557](https://jira.epic.oarcloud.noaa.gov/browse/UW-557) for factoring out some of the commonalities between the two MPAS drivers.